core/src/test/scala/org/broadinstitute/dsde/rawls/StatsDTestUtils.scala

```diff
@@ -0,0 +1,62 @@
+package org.broadinstitute.dsde.rawls
+
+import java.util.concurrent.TimeUnit
+
+import com.codahale.metrics.SharedMetricRegistries
+import com.codahale.metrics.health.SharedHealthCheckRegistries
+import com.readytalk.metrics.{StatsD, StatsDReporter}
+import org.broadinstitute.dsde.rawls.model.SubmissionStatuses.SubmissionStatus
+import org.broadinstitute.dsde.rawls.model.WorkflowStatuses.WorkflowStatus
+import org.broadinstitute.dsde.rawls.model.{Submission, Workspace, WorkspaceName}
+import org.mockito.Mockito.{atLeastOnce, inOrder => mockitoInOrder}
+import org.scalatest.concurrent.Eventually
+import org.scalatest.mock.MockitoSugar
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+
+/**
+  * Created by rtitle on 6/29/17.
+  */
+trait StatsDTestUtils { this: MockitoSugar with Eventually with RawlsTestUtils =>
+
+  protected def withStatsD[T](testCode: => T)(verify: Seq[(String, String)] => Unit = _ => ()): T = {
+    val statsD = mock[StatsD]
+    SharedMetricRegistries.clear()
+    SharedHealthCheckRegistries.clear()
+    val reporter = StatsDReporter.forRegistry(SharedMetricRegistries.getOrCreate("default"))
+      .convertRatesTo(TimeUnit.SECONDS)
+      .convertDurationsTo(TimeUnit.MILLISECONDS)
+      .build(statsD)
+    reporter.start(1, TimeUnit.SECONDS)
+    try {
+      val result = testCode
+      eventually(timeout(10 seconds)) {
+        val order = mockitoInOrder(statsD)
```

Comment:
Yes that's exactly right. Furthermore it checks that `connect()` and `close()` were called exactly once, and it waits up to 10 seconds (polling every x milliseconds, whatever the default is) before giving up.